pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,129 @@
+package agent
+
+import (
+	//corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Platform stores cluster configuration for an agent based install where hosts are booted and registered, ready
+// to be assigned and used in a cluster.
+type Platform struct {
+
+	// BareMetal contains bare metal specific configuration for this cluster.
+	BareMetal *BareMetal `json:"bareMetal"` // mandatory for now, possibly optional in future if other platforms are supported
+
+	// Networking is the configuration for the pod network provider in
+	// the cluster.
+	// TODO: should this be optional and defaulted?
+	Networking *Networking `json:"networking"`
```

Comment:
@avishayt @filanov thoughts on whether this networking is required or optional with defaulting?

Reply:
@abhinavdahiya if we send an install config without network cidrs set, do the defaults kick in and work?

Reply:
We're going to likely keep mandatory for now.